specs/lightclient/sync-protocol.md

```diff
@@ -0,0 +1,153 @@
+
+This document suggests a minimal light client design for the beacon chain that uses sync committees introduced in [this beacon chain extension](./beacon-chain.md).
+
+## Constants
+
+| Name | Value |
+| - | - |
+| `NEXT_SYNC_COMMITTEE_INDEX` | `IndexConcat(Index(BeaconBlock, 'state_root'), Index(BeaconState, 'next_sync_committee'))` |
+
+## Configuration
+
+### Misc
+
+| Name | Value |
+| - | - |
+| `MIN_SYNC_COMMITTEE_PARTICIPANTS` | `1` |
+| `MAX_VALID_LIGHT_CLIENT_UPDATES` | `uint64(2**64 - 1)` |
+
+### Time parameters
+
+| Name | Value | Unit | Duration |
+| - | - | :-: | :-: |
+| `LIGHT_CLIENT_UPDATE_TIMEOUT` | `Slot(2**13)` | slots | ~27 hours |
+
+## Containers
+
+#### `LightClientSnapshot`
+
+```python
+class LightClientSnapshot(Container):
+    # Beacon block header
+    header: BeaconBlockHeader
+    # Sync committees corresponding to the header
+    current_sync_committee: SyncCommittee
+    next_sync_committee: SyncCommittee
+```
+
+#### `LightClientUpdate`
+
+```python
+class LightClientUpdate(Container):
+    # Updated snapshot
+    snapshot: LightClientSnapshot
+    # Merkle branches for the next sync committee
+    next_sync_committee_branch: Vector[Bytes32, log2(NEXT_SYNC_COMMITTEE_INDEX)]
+    # Sync committee aggregate signature
+    sync_committee_bits: Bitlist[MAX_SYNC_COMMITTEE_SIZE]
+    sync_committee_signature: BLSSignature
+    # Fork version for the aggregate signature
+    fork_version
+```
+
+#### `LightClientStore`
+
+```python
+class LightClientStore(Container):
+    snapshot: LightClientSnapshot
+    valid_updates: List[LightClientUpdate, MAX_VALID_LIGHT_CLIENT_UPDATES]
+```
+
+## Light client state updates
+
+A light client maintains its state in a `store` object of type `LightClientStore` and receives `update` objects of type `LightClientUpdate`. Every `update` triggers `process_light_client_update(store, update, current_slot)` where `current_slot` is the currect slot based on some local clock.
+
+#### `is_valid_light_client_update`
+
+```python
+def is_valid_light_client_update(store: LightClientStore, update: LightClientUpdate) -> bool:
+    # Verify new slot is larger than old slot
+    old_snapshot = store.snapshot
+    new_snapshot = update.snapshot
+    assert new_snapshot.header.slot > old_snapshot.header.slot
+
+    # Verify update does not skip a sync committee period
+    old_period = compute_epoch_at_slot(old_snapshot.header.slot) // EPOCHS_PER_SYNC_COMMITTEE_PERIOD
+    new_period = compute_epoch_at_slot(new_snapshot.header.slot) // EPOCHS_PER_SYNC_COMMITTEE_PERIOD
+    assert new_period in (old_period, old_period + 1)
+
+    # Verify new snapshot sync committees
+    if new_period == old_period:
+        assert new_snapshot.current_sync_committee == old_snapshot.current_sync_committee
+        assert new_snapshot.next_sync_committee == old_snapshot.next_sync_committee
+    else new_period == old_period + 1:
```

Comment:
correct syntax here would be `elif`, but we can just use `else` given the prior assert on line 107.

```suggestion
    else:
```

Reply:
Thank you :)